services/loggers.ts

```diff
@@ -0,0 +1,116 @@
+import { TableService } from "azure-storage";
+import {
+  insertTableEntity,
+  ITableEntity
+} from "io-functions-commons/dist/src/utils/azure_storage";
+
+import { constVoid } from "fp-ts/lib/function";
+import { getRequiredStringEnv } from "io-functions-commons/dist/src/utils/env";
+import * as t from "io-ts";
+import { UTCISODateFromString } from "italia-ts-commons/lib/dates";
+
+const loggerService = new TableService(
+  getRequiredStringEnv("BONUS_STORAGE_CONNECTION_STRING")
+);
+
+// silently eat any exception
+const getTableLogger = <T>(
+  tableService: TableService,
+  tableName: string,
+  toEntity: (payload: T) => ITableEntity
+) => (payload: T) =>
+  insertTableEntity(tableService, tableName, toEntity(payload));
+
+/**
+ * Bind to a specific tracing function to create a wrapper for a fetch function that allow tracing of request and response.
+ * It returns a factory method that gets the original fetch function
+ *
+ * @param traceFn the tracing function
+ *
+ * @returns a factory method that gets the original fetch function and return an invariant version of fetch
+ */
+export const logHttpFetch = (
+  traceFn: (
+    input: RequestInfo,
+    init: RequestInit | undefined,
+    res: Response
+  ) => Promise<void>
+) => (f: typeof fetch): typeof fetch => {
+  return async (input, init) => {
+    const res = f(input, init);
+    res
+      .then(_ => _.clone())
+      .then(clonedRes => traceFn(input, init, clonedRes))
+      .catch(constVoid);
+    return res;
+  };
+};
+
+const IBasicHttpTrace = t.type({
+  RequestPayload: t.string,
+  ResponsePayload: t.string,
+  Timestamp: UTCISODateFromString
+});
+type IBasicHttpTrace = t.TypeOf<typeof IBasicHttpTrace>;
+
+export const createBasicHttpRequestTracer = <T extends IBasicHttpTrace>(
+  tableName: string
+) =>
+  getTableLogger<T>(loggerService, tableName, payload => ({
+    PartitionKey: payload.Timestamp.getTime().toString(),
```

Comment:
can we have a configurable PartitionKey in the tracer ? I mean the ability to pass a method that extracts it from the payload with a fallback on the timestamp if undefined. That would be useful to make an educate guess (a regex it's enough) on inps and ade payload to extract the fiscal code and use it as partition key

Reply:
both inps and ade are relying on this method because we added no specific logic, we're just tracing http request/response. If we add some logic (that is, we expect content to be in a certain format) we could be some tuning. But I'd rather duplicate this method instead of trying to find a size that fits all